File: src/main/java/commands/slashadapters/adapters/Waifu2xAdapter.java
package commands.slashadapters.adapters;

import commands.runnables.aitoyscategory.Waifu2xCommand;
import commands.slashadapters.Slash;
import commands.slashadapters.SlashMeta;
import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;

@Slash(command = Waifu2xCommand.class)
public class Waifu2xAdapter extends AIAdapterAbstract {

    @Override
    public SlashMeta process(SlashCommandEvent event) {
        return new SlashMeta(Waifu2xCommand.class, collectArgs(event));
    }

}

File: src/main/java/commands/slashadapters/adapters/Waifu2xAdapter.kt
package commands.slashadapters.adapters

import commands.runnables.aitoyscategory.Waifu2xCommand
import commands.slashadapters.Slash
import commands.slashadapters.SlashMeta
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent

@Slash(command = Waifu2xCommand::class)
class Waifu2xAdapter : AIAdapterAbstract() {

    override fun process(event: SlashCommandInteractionEvent): SlashMeta {
        return SlashMeta(Waifu2xCommand::class.java, collectArgs(event))
    }

}